### PR TITLE
Trainee homepage

### PR DESCRIPTION
**Pull request by yipingbenjaminliu**
Implementing the trainee landing page

### CODE REVIEW

File: ap/ap/views.py [hunk 1]
```diff
@@ -24,6 +25,13 @@ def home(request):
   if is_trainee(request.user):
     trainee = trainee_from_user(request.user)
     data['schedules'] = trainee.active_schedules
+
+    # check trainee term:
+    # if (first_year)
+      # calcFirstYearProgress
+    # elif (second_year)
+      # calcSecondYearProgress
+
```
Comment (yipingbenjaminliu) [inline ap/ap/views.py]: remove comments

File: ap/ap/views.py [hunk 2]
```diff
@@ -32,3 +40,21 @@ def home(request):
...
+  return (first_year_checked_list, int(float(first_year_progress) / 31102.0 * 100))
+
+
+def calcSecondYearProgress(user_checked_list):
+  second_year_checked_list = [int(book_code.split("_")[1]) for book_code in user_checked_list.keys() if book_code.startswith('2_')]
+
+  second_year_progress = 0
+  for checked_book in second_year_checked_list:
+    second_year_progress = second_year_progress + sum([int(chapter_verse_count) for chapter_verse_count in bible_books[checked_book][3]])
+  return (second_year_checked_list, int(float(second_year_progress) / 7957.0 * 100))
```
Comment (yipingbenjaminliu) [inline ap/ap/views.py]: try and refactor these two functions into a single one

**Comment by yipingbenjaminliu** [comment]
I've fully loaded the panels to make them dynamic. Few things left to do for the full implementation of the design

- [ ] individual cards needs to be properly styled
- [ ] services needs to be linked to back-end
- [ ] week select bar needs to be properly styled
bible reading select needs to be styled according to design including:
   - [ ] red over over box should make box border red
   - [ ] click on box should display dropdown
   - [ ] hover over option would make the option background red with text white
   - [ ] clicked option would populate box with option and make the option dark red with white text
- [ ] clicked on finalized would make background light blue and text white

all the design can be found on the dropbox
Dropbox\attendance project\djattendance\Design - Miriam\Pages\Trainee Homepage

File: ap/templates/index.html [hunk 3]
```diff
@@ -2,49 +2,230 @@
...
+          {% endfor %}
+          <span class="caret"></span>
+        </button>
+        <ul class="dropdown-menu" aria-labelledby="dropdownMenuButton">
+          {% for i, week in weeks %}
+          <li class="{% if i == current_week %}disabled{% endif %}"><a href="#"> {{week}} </a> </li>
+          {% endfor %}
+        </ul>
+      </div>
+      <div class="weekly_sum">
```
Comment (jontey) [inline ap/templates/index.html]: Is there a reason we are using divs instead of a table here?

**Pull request merged**